deploy/test-environments/fleet_api/src/api/common_api.py

```diff
@@ -142,3 +142,111 @@ def get_build_info(version: str, is_snapshot: bool) -> str:
+            url=url,
+        )
+        response_obj = munchify(response)
+        return response_obj.version
+
+    except APICallException as api_ex:
+        logger.error(
+            f"API call failed, status code {api_ex.status_code}. Response: {api_ex.response_text}",
+        )
+        return ""
+
+
+def get_cloud_security_posture_version(cfg: Munch, prerelease: bool = True) -> str:
+    """
+    Retrieve the version of the cloud_security_posture package.
+
+    Args:
+        cfg (Munch): Configuration object containing Kibana URL, authentication details, etc.
+        prerelease (bool, optional): Flag indicating whether to include
+                                        prerelease versions.Defaults to True.
+
+    Returns:
+        str: The version of the cloud_security_posture package, or None if the API call fails.
+
+    """
+    url = f"{cfg.kibana_url}/api/fleet/epm/packages"
+
+    request_params = {
+        "prerelease": prerelease,
+    }
+
+    try:
+        response = perform_api_call(
+            method="GET",
+            url=url,
+            auth=cfg.auth,
+            params={"params": request_params},
+        )
+        response_obj = munchify(response)
+        cloud_security_posture_version = None
+        for package in response_obj.response:
+            if package["name"] == "cloud_security_posture":
+                cloud_security_posture_version = package["version"]
+                # break
+
+        return cloud_security_posture_version
+    except APICallException as api_ex:
+        logger.error(
+            f"API call failed, status code {api_ex.status_code}. Response: {api_ex.response_text}",
+        )
+        return None
+
+
+def update_package_version(cfg: Munch, package_version: str):
+    """
+    Updates the version of the 'cloud_security_posture' package.
+
+    Args:
+        cfg (Munch): Configuration object containing Kibana URL, authentication details, etc.
+        package_version (str): The version to update the 'cloud_security_posture' package to.
+
+    Returns:
+        Union[Munch, str]: The response object if the API call is successful,
+        or an empty string on failure.
+
+    Raises:
+        APICallException: If the API call fails with an error.
+
+    """
+    # pylint: disable=duplicate-code
+    url = f"{cfg.kibana_url}/api/fleet/epm/packages/cloud_security_posture/{package_version}"
+    try:
+        response = perform_api_call(
+            method="POST",
+            url=url,
+            auth=cfg.auth,
+            params={
+                "json": {
+                    "force": True,
+                    "ignore_constraints": True,
+                },
+            },
+        )
+        response_obj = munchify(response)
+        return response_obj
+
+    except APICallException as api_ex:
+        logger.error(
+            f"API call failed, status code {api_ex.status_code}. Response: {api_ex.response_text}",
+        )
+        return ""
```

Comment:
why return a different object in case of an error? can we return an empty `Munch` or use dicts instead?

Reply:
also it seems like we don't use the returned object anywhere?